experimental/AlgebraicShifting/src/PartialShiftGraph.jl

```diff
@@ -0,0 +1,297 @@
+ [Set([2, 1]), Set([4, 1]), Set([2, 3]), Set([4, 2]), Set([4, 3])]
+```
+"""
+function partial_shift_graph(F::Field, complexes::Vector{T}, W::Union{WeylGroup, Vector{WeylGroupElem}};
+                             parallel::Bool = false,
+                             show_progress::Bool = true,
+                             task_size::Int=100) :: Tuple{Graph{Directed}, EdgeLabels, Vector{ComplexOrHypergraph}}  where T <: ComplexOrHypergraph;
+  # Deal with trivial case
+  if length(complexes) <= 1
+    return (graph_from_adjacency_matrix(Directed, zeros(length(complexes),length(complexes))), EdgeLabels())
+  end
+
+  # maybe we provide a flag to skip if the complexes are already sorted?
+  complexes = sort(complexes;lt=Oscar.isless_lex)
+
+  ns_vertices = Set(n_vertices.(complexes))
+  @req length(ns_vertices) == 1 "All complexes are required to have the same number of vertices."
+  n = collect(ns_vertices)[1]
+  
+  # inverse lookup K → index of K in complexes
+  complex_labels = Dict(Set(facets(K)) => index for (index, K) in enumerate(complexes))
+  
+  W2 = only(unique(parent.(W)))
+  rs_type = root_system_type(root_system(W2))
+  @req rs_type[1][1] == :A && rs_type[1][2] == n - 1 "Only Weyl groups type A_$(n-1) are currently support and received type $(T[1])."
+
+  phi = isomorphism(PermGroup, parent(first(W)))
+  # oscar runs tests in parallel, which can make pmap run in parallel even if parallel=false
+  # next line fixes this issue 
+  map_function = map
+  if parallel
+    # setup parallel parameters
+    channels = Oscar.params_channels(Union{PermGroup, Vector{SimplicialComplex}, Vector{UniformHypergraph}})
+    # setup parents needed to be sent to each process
+    Oscar.put_params(channels, codomain(phi))
+    map_function = pmap
+  end
+
+  if show_progress
+    edge_labels = reduce((d1, d2) -> mergewith!(vcat, d1, d2),
+                         @showprogress map_function(
+                           Ks -> multi_edges(F, phi.(W), Ks, complex_labels), 
+                           Iterators.partition(enumerate(complexes), task_size)))
+  else
+    edge_labels = reduce((d1, d2) -> mergewith!(vcat, d1, d2),
+                         map_function(
+                           Ks -> multi_edges(F, phi.(W), Ks, complex_labels), 
+                           Iterators.partition(enumerate(complexes), task_size)))
+  end
+  graph = graph_from_edges(Directed, [[i,j] for (i,j) in keys(edge_labels)])
+  return (graph, Dict(k => inv(phi).(v) for (k, v) in edge_labels), complexes)
+end
+
+function partial_shift_graph(F::Field, complexes::Vector{T};
+                             kwargs...) where T <: ComplexOrHypergraph
+  # Deal with trivial case
+  if length(complexes) <= 1
+    return (graph_from_adjacency_matrix(Directed, zeros(length(complexes),length(complexes))), EdgeLabels())
+  end
+
+  n = n_vertices(complexes[1])
+  W = weyl_group(:A, n - 1)
+  return partial_shift_graph(F, complexes, W; kwargs...)
+end
+
+@doc raw"""
```

Comment:
Would it make sense to refer to our preprint were this concept is introduced? (Same for partial_shift_graph)

Reply:
i've added a short sentence